core/src/ipld_ext.rs

```diff
@@ -0,0 +1,64 @@
+use anyhow::Result;
+use cid::Cid;
+use serde::Serialize;
+
+/// IpldExt is a trait for serialization, deserialization, CID calculation assuming dagcbor.
+pub trait IpldExt: Serialize {
```

Comment:
This name is not quite right... The Rust convention is that a `*Ext` trait is extending a trait with the * name. In this case it would be `SerializeExt`. Maybe we should rename it to `SerializeExt` and given its in `ceramic-core` its clear these are common serialization extensions on ceramic data types? I'd be happy with other names too but since there isn't an `Ipld` trait this is a bit confusing in its current name.

Reply:
I like `SerializeExt` that way we could add whatever other common Serializations we use.